packages/xo-web/src/xo-app/vm/tab-disks.js

```diff
@@ -69,34 +71,44 @@ import {
 
 const compareSrs = createCompare([isSrShared])
 
+@connectStore({
+  isAdmin,
+})
 class VdiSr extends Component {
   _getCompareContainers = createSelector(
     () => this.props.userData.vm.$pool,
-    poolId => createCompareContainers(poolId)
+    () => this.props.isAdmin,
+    (poolId, isAdmin) => (isAdmin ? createCompareContainers(poolId) : undefined)
   )
 
   _getSrPredicate = createSelector(
     () => this.props.userData.vm.$pool,
     poolId => sr => sr.$pool === poolId && isSrWritable(sr)
   )
 
-  _onChangeSr = sr => migrateVdi(this.props.item.vdi, sr)
+  _onChangeSr = sr =>
+    migrateVdi(this.props.item.vdi, sr, this.props.userData.resourceSet)
 
   render() {
-    const sr = this.props.item.vdiSr
+    const {
+      isAdmin,
+      item: { vdiSr },
+      userData: { resourceSet },
+    } = this.props
     return (
-      sr !== undefined && (
+      vdiSr !== undefined && (
         <XoSelect
           compareContainers={this._getCompareContainers()}
-          compareOptions={compareSrs}
+          compareOptions={isAdmin ? compareSrs : undefined}
           labelProp='name_label'
           onChange={this._onChangeSr}
           predicate={this._getSrPredicate()}
+          resourceSet={isAdmin ? undefined : resourceSet}
           useLongClick
-          value={sr}
-          xoType='SR'
+          value={vdiSr}
+          xoType={isAdmin ? 'SR' : 'resourceSetSr'}
         >
-          <Sr id={sr.id} link />
+          <Sr id={vdiSr.id} link={isAdmin} self={resourceSet !== undefined} />
```

Comment:
Why if a user isn't an admin and isn't in a `resourceSet` but has ACLs on the SR?

Reply:
Are you talking about the `link`?

Reply:
Yes

Reply:
This question can be also related to `xoType`.

Reply:
I'll fix it.